File: python/125. Valid Palindrome/solution.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        def clearInput(s: str) -> str:
            return re.sub('[^a-zA-Z0-9]', '', s).lower()

        cleared = clearInput(s)

        for i in range(len(cleared) // 2):
            if cleared[i] != cleared[-1 -i]:
                return False

        return True

File: java/125. Valid Palindrome/Solution.java
class Solution {
    public boolean isPalindrome(String s) {
        String cleared = clearInput(s);

        for (int i = 0; i < cleared.length() / 2; i++) {
            if (cleared.charAt(i) != cleared.charAt(cleared.length() - 1 - i)) return false;
        }
        return true;
    }

    public String clearInput(String s) {
        return s.replaceAll("[^a-zA-Z0-9]", "").toLowerCase();
    }
}